### PR TITLE
tests: use smoke test rather than test suite

### PR DESCRIPTION
**Pull request by vicentebolea**
Running the tests suite in the build step is an antipattern in CONDA, we spent time blacklisting tests here and there and the build takes significantly longer. Removed them and build/run the hello world smoke tests (c/C++/Python) in the test stage.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build-libadios.sh [hunk 1]
```diff
@@ -65,36 +54,16 @@ cmake               \
...
-    if [[ "${target_platform}" =~ osx ]]
-    then
-        exclude_tests+="|Test.Engine.DataMan1D.Serial"
-        exclude_tests+="|Test.Engine.DataMan1xN.Serial"
-        exclude_tests+="|Test.Engine.DataManSingleValues"
-        exclude_tests+="|Remote.BPWriteReadADIOS2stdio.GetRemote"
-        exclude_tests+="|Remote.BPWriteMemorySelectionRead.GetRemote"
-        exclude_tests+="|Remote.BPWriteMemorySelectionRead.FileRemote"
-    fi
+cmake --build build "-j$((CPU_COUNT*2))" -v
```
Comment (vicentebolea) [inline recipe/build-libadios.sh:65]: compilation process are also IO bounded.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% set name = "adios2" %}
 {% set version = "2.10.1" %}
 {% set sha256 = "ce776f3a451994f4979c6bd6d946917a749290a37b7433c0254759b02695ad85" %}
-{% set build = 0 %}
+{% set build = 1 %}
```
Comment (ax3l) [inline recipe/meta.yaml:4]: No build bump needed for this. We don't want to push a release for this alone.

**Review by ax3l — APPROVED** [review]
LGTM, thank you! Please do not bump the release, otherwise good to go.

File: recipe/build-libadios.sh (hunk 1 above)
Comment (ax3l) [reply recipe/build-libadios.sh:65]: I would not exceed this. Some (most) IO services check process utilization and oversubscription can lead to processes being killed sporadically.

**Pull request merged**